client/src/API.ts

```diff
@@ -7,76 +7,109 @@ import {
 } from "./Command";
 import { TextStyles } from "./components/Typewriter";
 
-interface SGProps {
-  handleConversation: Function;
+const BASE_URL = "http://127.0.0.1:5000/";
+
+export interface SGProps {
+  handleConversation: (text: string[], styles: TextStyles[]) => void;
   setGameID: Dispatch<SetStateAction<string>>;
-  isMounted: () => boolean;
+  handleUserInput: (input: string) => void;
 }
 
-export const startGame = async ({
-  handleConversation,
-  setGameID,
-  isMounted,
-}: SGProps) => {
-  const response = await fetch("http://127.0.0.1:5000/start");
+const fetchGame = async (
+  endpoint: string,
+  gameID: string,
+  userIn: string = ""
+) => {
+  return fetch(`${BASE_URL}${endpoint}${gameID}`, {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({ input: userIn }),
+  });
+};
 
-  if (!response.ok || !isMounted()) {
-    return;
+const getGameID = (): string => localStorage.getItem("game_id") as string;
+
+export const loadGame = async (props: SGProps) => {
+  const gameID = getGameID();
+  props.setGameID(gameID);
+  const loadedGame = await fetchGame("load/", gameID);
+
+  if (!loadedGame.ok) {
+    localStorage.removeItem("game_id");
+    return false;
   }
 
+  const data = await loadedGame.json();
+  data.response.forEach((response: any) => {
+    let { uText, uStyles } = processResponse(response);
+    uStyles = uStyles.map((style) => ({ ...style, characterDelayMs: 0 }));
+    props.handleConversation(uText, uStyles);
+  });
+  if (data.length && !data[data.length - 1].expects_user_input) {
+    props.handleUserInput("");
+  }
+};
+
+export const startGame = async (props: SGProps) => {
+  const response = await fetch(`${BASE_URL}start`);
```

Comment:
nit:: this looks a little weird to me because you put the trailing `/` in the `BASE_URL` string.